flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteInputChannel.java

```diff
@@ -237,6 +237,8 @@ void notifySubpartitionConsumed() {
 	@Override
 	void releaseAllResources() throws IOException {
 		if (isReleased.compareAndSet(false, true)) {
+			// stop advertising unannounced credit (best-effort)
+			getAndResetUnannouncedCredit();
```

Comment:
I would actually like to go one step further, because I don't like this part of the code in `CreditBasedPartitionRequestClientHandler` which is called for each credit announcement:
```
int unannouncedCredit = inputChannel.getAndResetUnannouncedCredit();
if (unannouncedCredit == 0 || inputChannel.isReleased()) {
```
-> I don't want it to access `AtomicInteger unannouncedCredit` and `AtomicBoolean isReleased` and it would be best it only one would suffice - or maybe this is (unnecessary) micro-optimisation?
I'll check whether my benchmarks would react on such a change...